crates/pypi-types/src/metadata.rs

```diff
@@ -350,9 +430,88 @@ mod tests {
 
         let s = "Metadata-Version: 2.3\nName: asdf\nVersion: 1.0\nRequires-Dist: foo";
         let meta = Metadata23::parse_pkg_info(s.as_bytes()).unwrap();
-        assert_eq!(meta.metadata_version, "2.3");
         assert_eq!(meta.name, PackageName::from_str("asdf").unwrap());
         assert_eq!(meta.version, Version::new([1, 0]));
         assert_eq!(meta.requires_dist, vec!["foo".parse().unwrap()]);
     }
+
+    #[test]
+    fn test_parse_pyproject_toml() {
+        let s = r#"
+            [project]
+            name = "asdf"
+        "#;
+        let meta = Metadata23::parse_pyproject_toml(s);
+        assert!(matches!(meta, Err(Error::FieldNotFound("version"))));
```

Comment:
That is even an invalid pyproject.toml ("The keys which are required but may be specified either statically or listed as dynamic are: version")